PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsJavaProjectWizard.java

```diff
@@ -202,18 +229,67 @@ public void configureJavaProject(String newProjectCompliance, IProgressMonitor m
 	        byte nSteps = 6;
 	        ((IProgressMonitor)monitor).beginTask(NewWizardMessages.JavaCapabilityConfigurationPage_op_desc_java, nSteps);
 	        try {
-	        	IProject project = addHDInsightNature(monitor);
+	            IProject project = addHDInsightNature(monitor);
 	            Method method = JavaCapabilityConfigurationPage.class.getDeclaredMethod("getBuildPathsBlock");
 	            method.setAccessible(true);
 	            Object r = method.invoke(this);
 	            ((BuildPathsBlock) r).configureJavaProject(newProjectCompliance, new SubProgressMonitor((IProgressMonitor)monitor, 5));
+		            
+	            addMoreSourcetoClassPath();
 	        } catch (OperationCanceledException | NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException | InvocationTargetException ex) {
 	            throw new InterruptedException();
 	        } finally {
 	            ((IProgressMonitor)monitor).done();
 	        }
 	    }
 
+		private void addMoreSourcetoClassPath() throws JavaModelException {
+			HDInsightJavaPageOne previousPage = (HDInsightJavaPageOne)getPreviousPage();
+			if (previousPage.sparkLibraryOptionsPanel.useMaven) {
+				IProject project = this.getJavaProject().getProject();
+				final IFolder sourceRootFolder = project.getFolder("src");
+				if (!sourceRootFolder.exists()) {
+					try {
+						sourceRootFolder.create(false, true, null);
+					} catch (CoreException e) {
+						e.printStackTrace();
```

Comment:
The same as above

Reply:
Fixed

---
In reply to: [136775434](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136775434) [](ancestors = 136775434)